AWS: update reference to auto-retrying ResourceNotFound (#72314)
ansible/ansible: docs/docsite/rst/dev_guide/platforms/aws_guidelines.rst

           module.fail_json_aws(e, msg="Could not describe some resource")

Prior to Ansible 2.10 if the underlying ``describe_some_resources`` API call threw
a ``ResourceNotFound`` exception, ``AWSRetry`` would take this as a cue to retry until

> it's not thrown (this is so that when creating a resource, we can just retry until it

Reviewer: ```suggestion
it is not thrown (this is so that when creating a resource, we can just retry until it
```

> exists).  This default was changed and it's now necessary to explicitly request

Reviewer: ```suggestion
exists).  This default was changed and it is now necessary to explicitly request
```

@tremble Do you think it would be better to specify a particular version of the Ansible or AWS collection here?

Author: I think it pre-dates the start of the collection...

this behaviour.  This can be done by using the ``catch_extra_error_codes``
argument on the decorator.

.. code-block:: python

   @AWSRetry.exponential_backoff(retries=5, delay=5, catch_extra_error_codes=['ResourceNotFound'])

>    def describe_some_resource_with_backoff(client, **kwargs):

Reviewer 2: `describe_some_resource_until_it_exists` now?

Author: ```suggestion
   def describe_some_resource_retry_missing(client, **kwargs):
```
